.github/workflows/e2e-fork.yml

```diff
@@ -51,3 +51,19 @@ jobs:
         run: |
           cd e2e
           make e2e-test test=${{ matrix.test }}
+
+  # e2e-fork-success is a dummy job that is used to tally the results of the e2e-fork job. The name of this job
+  # can be used in the branch protection rules, as a single target, to make sure all tests pass before merging.
+  # If e2e-fork is skipped, e2e-fork-success will be skipped as well.
```

Comment:
More info: if `e2e-fork` runs we use the output from the job (via the `need.*.result`) to check that each of the matrix jobs generated didn't fail, get cancelled or skipped.

If `e2e-fork` doesn't run, the `needs: e2e-fork` directive will mark `e2e-fork-success` as skipped.